common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1481,6 +1496,9 @@ public void onComplete(String streamId) throws IOException {
           // the client in cases of duplicate even though no data is written.
```

Comment:
`updateIgnoredBlockBytes` for `isStale` and `isTooLate` conditions above ?

Reply:
We throw the exception so `onFailure` will be called and they will be updated there.

Reply:
It need not necessarily be detected during onData - it can go stale subsequently as well.
(Else we wont need these two conditions here at all :-) )

Reply:
The UTs `testBlockPushWithOlderShuffleMergeId` and `testFinalizeWithOlderShuffleMergeId ` have some stale blocks. Before applying updateIgnoredBlockBytes for isStale and isTooLate conditions above, they are not taken account into the ignored bytes. They are able to be captured after. So I think we should update ignored block bytes there. Updated the PR, thanks.

Reply:
@mridulm I didn't get your point. I thought you pointed that in `onComplete` when we detect `tooLate` or `tooStale`, we should update the metric. However, when we detect `tooLate` or `tooStale` we throw an exception, so that will trigger `onFailure`.

@rmcyang Can you check if those tests trigger `onFailure` after `onComplete` throws an exception? If they don't then that's why the metric didn't get updated.

Reply:
>  However, when we detect tooLate or tooStale we throw an exception, so that will trigger onFailure.

I see what you mean, I read it as this wont happen and should have triggered onFailures from onData directly.

Agree - since `TransportRequestHandler.processStreamRequest` is explicitly doing the delegation to `onFailures` when `onComplete` fails, that should be fine.

Reply:
@otterc I checked those UTs. We didn't call onFailure explicitly for those steams, I have explicitly call onFailure now and the ignored bytes metric is able to be updated.